tools/travis/test_openwhisk.sh

```diff
@@ -2,57 +2,87 @@
 
 set -e
 
-cd $TRAVIS_BUILD_DIR
-./tools/travis/scancode.sh
-make lint
-make build
-make test
-export PATH=$PATH:$TRAVIS_BUILD_DIR;
-make native_test;
-
-export OPENWHISK_HOME="$(dirname "$TRAVIS_BUILD_DIR")/incubator-openwhisk";
+#
+#  At this point, the Travis build should already have built the binaries and
+#  the release.  If you're running manually, this command should get you to
+#  the same place:
+#
+#    ./gradlew buildBinaries release --PcrossCompile=true
+#
+#  Also at this point, you should already have incubator-openwhisk pulled down
+#  from gradle in the parent directory, using a command such as:
+#
+#    git clone --depth 3 https://github.com/apache/incubator-openwhisk.git
+#
+#  To be clear, your directory structure will look something like...
+#
+#      $HOMEDIR
+#       |- incubator-openwhisk
+#       |- incubator-openwhisk-cli (This project)
+#       |- incubator-openwhisk-utilities (For scancode)
+#
+#  The idea is to only build once and to be transparent about building in
+#  the Travis script.  To that end, some of the other builds that had been
+#  done in this script will be moved into Travis.yml.
+#
+
+#
+#  Figure out default directories, etc., so we're not beholden to Travis
+#  when running tests of the script.
+#
+scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+TRAVIS_BUILD_DIR="$( cd "${TRAVIS_BUILD_DIR:-$scriptdir/../..}" && pwd )"
+export TRAVIS_BUILD_DIR
+
+# For the gradle builds.
 HOMEDIR="$(dirname "$TRAVIS_BUILD_DIR")"
-cd $HOMEDIR
+OPENWHISK_HOME="$( cd "${OPENWHISK_HOME:-$HOMEDIR/incubator-openwhisk}" && pwd )"
+export OPENWHISK_HOME
 
-# Clone the OpenWhisk code
-git clone --depth 3 https://github.com/apache/incubator-openwhisk.git
+#
+#  These are the basic tests
+#
+../incubator-openwhisk-utilities/scancode/scanCode.py $TRAVIS_BUILD_DIR
 
-# Build script for Travis-CI.
-WHISKDIR="$HOMEDIR/incubator-openwhisk"
+#  Run separate test scopes as separate
+./gradlew --console=plain goLint
+./gradlew --console=plain goTest -PgoTags=unit
+export PATH=$PATH:$TRAVIS_BUILD_DIR
+./gradlew --console=plain goTest -PgoTags=native
 
-cd $WHISKDIR
+#
+#  Set up the OpenWhisk environment ( TODO: reusable script for incubtor-openwhisk? )
+cd $OPENWHISK_HOME
 ./tools/travis/setup.sh
 
 ANSIBLE_CMD="ansible-playbook -i environments/local -e docker_image_prefix=testing"
-TERM=dumb ./gradlew distDocker -PdockerImagePrefix=testing
+./gradlew --console=plain distDocker -PdockerImagePrefix=testing
```

Comment:
Should this be `buildBinaries` instead of `distDocker`?

Reply:
Why do we need to build again here, shouldn't the binaries already exist from the build?

Reply:
Check the working directory.  This is actually a build of OpenWhisk (not the CLI) for integration so the integration environment is running.  So this really is a build of the docker images OpenWhisk uses to run its cluster.